# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.h
# Commit: 80193c01e94b
# Full Hash: 80193c01e94bbd93990445cacc7943af1b935cba
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2024-06-25 09:36:02
# Regressor Bug: 1728331
# File Overlap Count: 1
# Description:
#   Bug 1728331 - Part 1: Remove e10s process recycling, r=smaug
#   
#   This logic did nothing in Fission, however it did have an effect for
#   e10s-Android, which will no longer have process recycling. I expect the impact
#   of this to be minimal given Android's tendency to kill processes already.
# ==============================================================================

diff -r a8a15b242d9d -r 80193c01e94b dom/ipc/ContentParent.h
--- a/dom/ipc/ContentParent.h	Mon Jun 24 23:19:27 2024 +0000
+++ b/dom/ipc/ContentParent.h	Mon Jun 24 23:19:27 2024 +0000
@@ -686,16 +686,6 @@
       sBrowserContentParents;
   static mozilla::StaticAutoPtr<LinkedList<ContentParent>> sContentParents;
 
-  /**
-   * In order to avoid rapidly creating and destroying content processes when
-   * running under e10s, we may keep alive a single unused "web" content
-   * process if it previously had a very short lifetime.
-   *
-   * This process will be re-used during process selection, avoiding spawning a
-   * new process, if the "web" remote type is being requested.
-   */
-  static StaticRefPtr<ContentParent> sRecycledE10SProcess;
-
   void AddShutdownBlockers();
   void RemoveShutdownBlockers();
 
@@ -759,24 +749,6 @@
   void ForwardKnownInfo();
 
   /**
-   * We might want to reuse barely used content processes if certain criteria
-   * are met.
-   *
-   * With Fission this is a no-op.
-   */
-  bool TryToRecycleE10SOnly();
-
-  /**
-   * If this process is currently being recycled, unmark it as the recycled
-   * content process.
-   * If `aForeground` is true, will also restore the process' foreground
-   * priority if it was previously the recycled content process.
-   *
-   * With Fission this is a no-op.
-   */
-  void StopRecyclingE10SOnly(bool aForeground);
-
-  /**
    * Removing it from the static array so it won't be returned for new tabs in
    * GetNewOrUsedBrowserProcess.
    */